# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: 6c61c412fe0b
# Full Hash: 6c61c412fe0b1afbec5d58dd6fea7c76d0531ff6
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Don't use DOM objects internally in Web Codecs implementation. r=chunmin
#   
#   Depends on D181328
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D182476
# ==============================================================================

diff -r 7dd351fce5ad -r 6c61c412fe0b dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:14 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:15 2023 +0000
@@ -47,10 +47,10 @@
 Result<Span<uint8_t>, nsresult> GetSharedArrayBufferData(
     const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aBuffer);
 
-Result<OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer, nsresult>
-CloneBuffer(
+Result<Ok, nsresult> CloneBuffer(
     JSContext* aCx,
-    const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aBuffer);
+    OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aDest,
+    const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aSrc);
 
 /*
  * The following are utilities to convert between VideoColorSpace values to
